extensions/src/main/java/com/redhat/lightblue/extensions/sequance/Sequance.java

```diff
@@ -0,0 +1,25 @@
+package com.redhat.lightblue.extensions.sequance;
+
+import com.redhat.lightblue.metadata.SimpleField;
+
+/**
+ * TODO: Does SimpleField have all the info required by the backend to manage sequences?
+ *
+ * @author mpatercz
+ *
+ */
+public interface Sequance {
+
+    public void createSequance(SimpleField field);
+
+    public void removeSequance(SimpleField field);
+
+    /**
+     * Generate next value in sequance. This should be handled automatically by the backend on insert.
+     *
+     * @param field
+     * @return
+     */
+    public Long nextInSequance(SimpleField field);
```

Comment:
You should use a Type object to describe the type of the generated
sequence. Then you can use type.toJson(Object) to add that to a json
document.

On Tue, Aug 18, 2015 at 2:54 AM, Marek <notifications@github.com> wrote:

> In
> extensions/src/main/java/com/redhat/lightblue/extensions/sequance/Sequance.java
> <https://github.com/lightblue-platform/lightblue-core/pull/431#discussion_r37277131>
> :
>
> > + * @author mpatercz
> > + *
> > + */
> > +public interface Sequance {
> > +
> > +    public void createSequance(SimpleField field);
> > +
> > +    public void removeSequance(SimpleField field);
> > +
> > +    /**
> > +     * Generate next value in sequance. This should be handled automatically by the backend on insert.
> > +     *
> > +     * @param field
> > +     * @return
> > +     */
> > +    public Long nextInSequance(SimpleField field);
>
> How would a sequence of Objects look like? I wouldn't add it just to have
> to cast it to BigInteger/Long.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/431/files#r37277131>
> .
>